paprika/align.py

```diff
@@ -192,3 +192,158 @@ def offset_structure(structure, offset):
+    for atom in range(len(structure.atoms)):
+        aligned_coords[atom] = structure.coordinates[atom] - centroid * dim_mask
+    structure.coordinates = aligned_coords
+
+    return structure
+
+
+def align_principal_axes(structure, atom_mask=None, princ_axis=None, v_axis=None):
+    """Aligns the a chosen principal axis of a system to a specified axis. 
+       This function is based on the method given in the link:
+
+        * https://www.ks.uiuc.edu/Research/vmd/script_library/scripts/orient/
+
+    Parameters
+    ----------
+    structure : parmed.Structure
+        Molecular structure containing coordinates.
+    atom_mask : str
+        A mask that filter specific atoms for calculating the moment of inertia.
+    princ_axis : int
+        Which principal axis to align? (choice: 1, 2 or 3, default: 1).
+    v_axis: list
+        The axis vector to align the system to (default: [0, 0, 1]).
+
+    Returns
+    -------
+    structure : parmed.Structure
+        A molecular structure with a principal axis aligned to a vector.
+
+    Examples
+    --------
+    The commands below mimics the example given in the link above for VMD.
+
+        >>> structure = align_principal_axes(structure, princ_axis=0, v_axis=[0,0,1])
+        >>> structure = align_principal_axes(structure, princ_axis=1, v_axis=[0,1,0])
+
+    """
+    # Check princ_axis
+    if princ_axis is None:
+        princ_axis = 1
+    elif princ_axis not in [0, 1, 2]:
+        raise SystemExit(
+            'Error: "princ_axis" can only be an integer value of 0, 1 or 2.'
+        )
+
+    # Axis vector for alignment
+    if v_axis is None:
+        v_axis = [0, 0, 1]
+    elif len(v_axis) != 3:
+        raise SystemExit('Error: "v_axis" must be a list with 3 elements, e.g. [0,0,1]')
+    v_axis = np.array(v_axis)
+
+    # Get coordinates and masses
+    coordinates = structure.coordinates
+    masses = np.asarray([atom.mass for atom in structure.atoms])
+    if atom_mask:
+        coordinates = structure[atom_mask].coordinates
+        masses = np.asarray([atom.mass for atom in structure[atom_mask].atoms])
+
+    # Calculate center of mass
+    centroid = pmd.geometry.center_of_mass(coordinates, masses)
+
+    # Construct Inertia tensor
+    Ixx = Ixy = Ixz = Iyy = Iyz = Izz = 0
+    for xyz, mass in zip(coordinates, masses):
+        xyz -= centroid
+        Ixx += mass * (xyz[1] * xyz[1] + xyz[2] * xyz[2])
+        Ixy -= mass * (xyz[0] * xyz[1])
+        Ixz -= mass * (xyz[0] * xyz[2])
+        Iyy += mass * (xyz[0] * xyz[0] + xyz[2] * xyz[2])
+        Iyz -= mass * (xyz[1] * xyz[2])
+        Izz += mass * (xyz[0] * xyz[0] + xyz[1] * xyz[1])
+
+    inertia = np.array([[Ixx, Ixy, Ixz], [Ixy, Iyy, Iyz], [Ixz, Iyz, Izz]])
+
+    # Principal axis
+    evals, evecs = np.linalg.eig(inertia)
+    evecs = evecs[:, evals.argsort()]  # <-- Numpy does not sort the vectors properly
+    p_axis = evecs[:, princ_axis]
+
+    # Calculate Rotation matrix
```

Comment:
Another reason why I think you can merge with https://github.com/slochower/pAPRika/blob/1bfce3a2705f908b354a5a2b0c2bc5261e767cc8/paprika/align.py#L70-L79

Reply:
Refactored code